src/Message.ts

```diff
@@ -134,18 +134,51 @@ export default class DicomMessage {
     var stream = new ReadBufferStream(buffer),
       useSyntax: NormalizedSyntax = EXPLICIT_LITTLE_ENDIAN;
     stream.reset();
+    // Skip the File Preamble (128 bytes)
     stream.increment(128);
+    // Check the DICOM Prefix
     if (stream.readString(4) != "DICM") {
       throw new Error("Invalid: not a dicom file");
     }
-    var el = DicomMessage.readTag(stream, useSyntax),
+    var el = DicomMessage.readTag(stream, useSyntax);
+
+    var metaLength
+
+    // File Meta Information Group Length is correctly suplied
+    if (el.tag.value == 0x00020000) {
       metaLength = el.values[0];
+    // No Length but there are metadata tags
+    } else if (el.tag.value>>16 == 2) {
```

Comment:
```suggestion
    } else if ((el.tag.value & 0xFFFF0000) === 0x00020000) {
```

This may seem like a little bit of a nitpick but it's a lot clearer that we're just masking off the lower 16 bits like this (and makes the relationship to the value we compare with on L148 clearer).

Reply:
I'm pretty sure I had some version of this along the way, I don't mind either way.

Reply:
Yepyep. There's still one instance of this comparison (on this line).